Add doc requirements to PR template

meker12 opened the pull request:
# Description

Add release note and documentation requirements to PR template


### Fixed Issues (if relevant)

n/a

### Manual testing scenarios
n/a

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [ ] Pull request introduces user-facing changes and includes meaningful release notes and documentation
 - [ ] All commits are accompanied by meaningful commit messages

meker12 committed:
Add doc requirements to PR template

NadiyaS commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -30,8 +30,17 @@
+ - [ ] Pull request introduces user-facing changes and includes meaningful release notes and documentation
  - [ ] All commits are accompanied by meaningful commit messages
- - [ ] All new or changed code is covered with unit/integration tests (if applicable)
- - [ ] All automated tests passed successfully (all builds on Travis CI are green)
```
Why we removed these 2 items?

meker12 replied:
Restored these items. This was a copy/paste error.  The PR templates in the Cloud docker, patches, and components repositories do not have these items in the checklist.  Do they need to be added to those PR templates?

NadiyaS replied:
It's OK not to have it for docker PRs and MCC. But steps for manual tests should be required

BarnyShergold commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -30,8 +30,17 @@
+Add link to Magento DevDocs PR or Issue, if needed.
+
 ### Contribution checklist
  - [ ] Pull request has a meaningful description of its purpose
```
Please can we have this clarified that it is not sufficient to just put a link to a JIRA ticket as the description? This is often done and the issue is that Community Maintainers like myself do not have access to these tickets.

meker12 replied:
I added clarification in the Description comments.

meker12 committed:
Require text description for PRs

- Added requirement to provide text description for PR since some PRs have only a link to an MCLOUD JIRA ticket,
and that information is not accessible to community

meker12 commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -30,8 +32,17 @@
+
 ### Contribution checklist
  - [ ] Pull request has a meaningful description of its purpose
+ - [ ] Pull request introduces user-facing changes and includes meaningful release notes and documentation
```
```suggestion
 - [ ] Pull request introduces user-facing changes and includes meaningful updates for any required release notes and documentation changes
```

meker12 committed:
Update .github/PULL_REQUEST_TEMPLATE.md

meker12 committed:
Restore test items to checklist

shiftedreality committed:
Merge branch 'develop' into mae-pr-template-update-for-doc-reqs

shiftedreality commented:
@BarnyShergold, please approve

ghost commented:
Hi @meker12, thank you for your contribution! </br>Please, complete [Contribution Survey](https://opensource.magento.com/survey/contribution/magento+ece-tools+739), it will take less than a minute. </br>Your feedback will help us to improve contribution process.
